compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -544,9 +546,15 @@ checkValidProjectionsKey = \case
   expr ->
     throwWithContext (EInvalidKeyExpression expr)
 
-checkTemplateKey :: MonadGamma m => ExprVarName -> Qualified TypeConName -> Maybe TemplateKey -> m ()
-checkTemplateKey param tcon mbKey =
-  for_ mbKey $ \TemplateKey{..} -> do
+checkSupportsContractKeys :: MonadGamma m => m ()
+checkSupportsContractKeys = do
```

Comment:
Surprised there isn't a generate `checkSupported` which takes the predicate an error message and hides the magic.